src/senaite/patient/content/patient.py

```diff
@@ -0,0 +1,265 @@
+                return
+
+        patient = patient_api.get_patient_by_mrn(
+            data.mrn, full_object=False, include_inactive=True)
+        if patient:
+            raise Invalid(_("Patient Medical Record # must be unique"))
+
+    @invariant
+    def validate_email(data):
+        """Checks if the email is correct
+        """
+        if not data.email:
+            return
+        if not is_valid_email_address(data.email):
+            raise Invalid(_("Patient email is invalid"))
+
+
+@implementer(IPatient)
+class Patient(Item):
+    """Results Interpretation Template content
+    """
+
+    def Title(self):
+        fullname = self.get_fullname()
+        return fullname.encode("utf8")
+
+    def get_mrn(self):
+        return self.mrn
+
+    def set_mrn(self, value):
+        value = value.strip()
+        if self.mrn == value:
+            # noting changed
+            return
+        if patient_api.get_patient_by_mrn(
+                value, full_object=False, include_inactive=True):
+            raise ValueError("A patient with that MRN already exists!")
+        self.mrn = value
+
+    def get_fullname(self):
+        fullname = self.fullname
+        if not fullname:
+            return ""
+        return fullname.strip()
+
+    def set_fullname(self, value):
+        if not isinstance(value, string_types):
+            self.fullname = ""
+        self.fullname = api.safe_unicode(value)
+
+    def get_email(self):
+        email = self.email
+        if not email:
+            return u""
+        return email.strip()
+
+    def get_gender(self):
+        genders = dict(GENDERS)
+        return genders.get(self.gender)
+
+    def set_gender(self, value):
+        for k, v in GENDERS:
+            if value == v:
+                value = k
+        self.gender = value
+
+    def get_birthdate(self):
+        if not self.birthdate:
+            return None
+        return self.birthdate
+
+    def set_birthdate(self, value):
+        """Set birthdate field
+
+        Tries to convert value to datetime before set.
+        """
+        if isinstance(value, DateTime):
+            # convert DateTime -> datetime
+            value = value.asdatetime()
+        elif isinstance(value, string_types):
+            # convert string to datetime
+            dt = api.to_date(value, None)
+            if dt is not None:
+                value = dt.asdatetime()
+        # Ensure the datetime object has no timezone set
+        # Happens when setting `DateOfBirth` of the Sample to the Patient
+        #
+        # TypeError: can't compare offset-naive and offset-aware datetimes
+        if isinstance(value, datetime):
+            value = value.replace(tzinfo=None)
+        else:
+            value = None
+        self.birthdate = value
+
+    def get_age(self):
+        """Calculate the Age from the birthdate
+        """
+        birthdate = self.birthdate
+        if not birthdate:
+            return None
+        now = datetime.now()
```

Comment:
Important to not mix the age of the patient (at current time) with the age of the patient when the sample was collected. In sample context, the age should always refer to the age of the patient when the sample was collected